spmv/Matrix.cpp

```diff
@@ -29,6 +82,15 @@ Matrix<T>::~Matrix()
 #endif
 }
 
+template <typename T>
+size_t Matrix<T>::size() const
```

Comment:
We should probably add some documentation... 
I wouldn't expect this function to return the size in bytes .